File: StarcevMatvey/Term 2/Block_3/Task_1/Bots/RandomBot.cs
using BlackJackEnumerations;
using BlackJack;

namespace Bots
{
    public class RandomBot : Player
    {
        private Random random;

        public RandomBot(string name, int balance) : base(name, balance)
        {
            random = new Random(DateTime.Now.Millisecond);
        }

        public override int GetNewBet()
        {
            return random.Next(Balance);
        }

        public override PlayerMove GetMove()
        {
            return (PlayerMove)random.Next(3);
        }

    }
}

File: IvanShurenkov/Second Term/Block2/Task1/lib/random_bot/src/main/kotlin/lib/blackjack/bots/RandomBot.kt
package lib.blackjack.bots

import lib.blackjack.base.state.GameInfo
import lib.blackjack.base.Hand
import lib.blackjack.base.IPlayer
import lib.blackjack.base.state.PlayerMove
import lib.blackjack.base.state.PlayerMove.HIT
import lib.blackjack.base.state.PlayerMove.STAND
import kotlin.random.Random.Default.nextBoolean

object RandomBot : IPlayer {
    override val name: String = "RandomBot"

    override fun getBet(bankroll: UInt, gameInfo: GameInfo): UInt {
        return gameInfo.rangeBet.first
    }

    override fun getMove(hand: Hand, gameInfo: GameInfo): PlayerMove {
        return when {
            nextBoolean() -> HIT
            else -> STAND
        }
    }
}